ethereum_history_api/circuits/lib/src/log_test.nr

```diff
@@ -0,0 +1,78 @@
+mod extract_log {
+    use crate::fixtures::mainnet::{
+        cancun::{
+        small_block::{receipt::receipt_rlp as small_block_receipt_rlp, log::log as small_block_log},
+        approve::{receipt::receipt_rlp as approve_receipt_rlp, log::log as approve_log}
+    }
+    };
+    use crate::log::{Log, extract_log, PhantomMaxLogDataLen, PhantomMaxLogsCount};
+    use crate::misc::arrays::sub_array_equals_up_to_length;
+
+    global phantom_max_logs_count: PhantomMaxLogsCount<8> = PhantomMaxLogsCount {};
+
+    #[test]
+    fn success_small_block() {
+        let log = extract_log(small_block_receipt_rlp, 0, phantom_max_logs_count);
+        assert_eq(log, small_block_log);
+    }
+
+    #[test]
+    fn success_approve() {
+        let log = extract_log(approve_receipt_rlp, 0, phantom_max_logs_count);
+        assert_eq(log, approve_log);
+    }
+
+    #[test]
+    fn big_max_log_data_len() {
+        let log: Log<100> = extract_log(small_block_receipt_rlp, 0, phantom_max_logs_count);
```

Comment:
Let's extract those consts 100 and 32 into MAX_LOG_DATA_LEN_M and MAX_LOG_DATA_LEN_L so that it's more visible